### PR TITLE
QA-84: Support trigger builds on GitLab from release_tool.py

### PR DESCRIPTION
**Pull request by lluiscampos**
```
The tool, by default, will still trigger the builds on Jenkins CI, while
this behaviour is now configurable using -c cli parameter.

The implementation basically follows the same logic as for Jenkins, only
deciding in which server to trigger the build in the last stage. One
notably difference is that the default params are fetched from the API
for Jenkins but parsed from .gitlab-ci.yaml for GitLab.

Changelog: None

Signed-off-by: Lluis Campos <lluis.campos@northern.tech>
```

### CODE REVIEW

File: extra/release_tool.py [hunk 1]
```diff
@@ -35,13 +35,22 @@
 - JENKINS_USER / JENKINS_PASSWORD environment variables
 - 'pass' password management storage."""
 
+GITLAB_SERVER = "https://gitlab.com/api/v4"
+GITLAB_JOB = "projects/Northern.tech%2FMender%2Fmender-qa"
+GITLAB_TOKEN = None
+GITLAB_PASSWORD = None
```
Comment (lluiscampos) [inline extra/release_tool.py]: This is unused. I'll remove

**Comment by lluiscampos** [comment]
Thank you for your review. Merging!

File: extra/release_tool.py [hunk 2]
```diff
@@ -35,13 +35,21 @@
...
+- GITLAB_TOKEN environment variable"""
+
 # What we use in commits messages when bumping versions.
 VERSION_BUMP_STRING = "Bump versions for Mender"
 
 # Whether or not pushes should really happen.
 PUSH = True
 # Whether this is a dry-run.
 DRY_RUN = False
+# Wheter we are using GitLab
```
Comment (kacf) [inline extra/release_tool.py]: Nitpick: `s/Wheter/Whether/`

File: extra/release_tool.py [hunk 3]
```diff
@@ -1317,6 +1376,44 @@ def trigger_jenkins_build(state, tag_avail):
...
+    init_gitlab_creds()
+    if not GITLAB_TOKEN:
+        raise SystemExit(GITLAB_CREDS_MISSING_ERR)
+
+    headers = {'PRIVATE-TOKEN': GITLAB_TOKEN}
+
+    # Prepare json POST data
+    # See https://docs.gitlab.com/ee/api/pipelines.html#create-a-new-pipeline
+    # mender-qa ref hardcoded to master, it will not work for PR-like refs
+    postdata = {"ref": "master", "variables": []}
```
Comment (kacf) [inline extra/release_tool.py]: It should be pretty easy to make a special case for `MENDER_QA_REV` and `ref` here, don't you think? I'm using PRs with mender-qa pretty often.

Comment (lluiscampos) [reply extra/release_tool.py]: Yep, you are right. Thanks for doing this in PR #673

**Pull request merged**